src/main/java/org/jvnet/hudson/test/RealJenkinsRule.java

```diff
@@ -441,6 +444,18 @@ public RealJenkinsRule includeTestClasspathPlugins(boolean includeTestClasspathP
         return this;
     }
 
+    /**
+     * Allows JENKINS_HOME initialization to be delayed until {@link #startJenkins} is called for the first time.
+     * <p>
+     * This allows methods such as {@link #addPlugins} to be called dynamically inside of test methods, which enables
+     * related tests that need to configure {@link RealJenkinsRule} in different ways to be defined in the same class
+     * using only a single instance of {@link RealJenkinsRule}.
+     */
+    public RealJenkinsRule withLazyProvisioning(boolean lazyProvisioning) {
```

Comment:
I don't particularly care about this name and it is somewhat verbose. Perhaps just `lazy` would be good enough.

Reply:
`prepareHomeLazily`? No strong opinion.

Reply:
https://github.com/jenkinsci/jenkins-test-harness/pull/690/commits/1dcdc9ef15a957cccaab772943e18d7f641e69b1